flight/src/conversion.rs

```diff
@@ -0,0 +1,159 @@
+use crate::types::*;
+use anyhow::Result;
+use arrow::array::{
+    ArrayRef, BinaryBuilder, ListBuilder, PrimitiveBuilder, StringBuilder, StructArray,
+    UInt8Builder,
+};
+use arrow::datatypes::{DataType, Field};
+use arrow::record_batch::RecordBatch;
+use std::sync::Arc;
+
+#[derive(Debug)]
+pub struct ConclusionEventBuilder {
+    event_type: UInt8Builder,
+    stream_id: BinaryBuilder,
+    stream_type: UInt8Builder,
+    controller: StringBuilder,
+    data: BinaryBuilder,
+    previous: ListBuilder<BinaryBuilder>,
+}
+
+impl Default for ConclusionEventBuilder {
+    fn default() -> Self {
+        Self {
+            event_type: PrimitiveBuilder::new(),
+            stream_id: BinaryBuilder::new(),
+            stream_type: PrimitiveBuilder::new(),
+            controller: StringBuilder::new(),
+            data: BinaryBuilder::new(),
+            previous: ListBuilder::new(BinaryBuilder::new())
+                .with_field(Field::new_list_field(DataType::Binary, false)),
+        }
+    }
+}
+
+impl ConclusionEventBuilder {
+    fn append(&mut self, event: &ConclusionEvent) {
+        match event {
+            ConclusionEvent::Data(data_event) => {
+                self.event_type.append_value(0);
```

Comment:
Can we add a method on ConclusionEvent that can report the type as an integer? This way the mapping of variant to integer type is explicit on the type instead of hidden in the conversion implementation.